### ISSUE TITLE
capture_serverless makes lambda fail when input is not a JSON-dict

### ISSUE THREAD

**Issue by fjoanr** [elastic/apm-agent-python#1746]
**Describe the bug**: 

Hi! 

I have encountered a bug when using `@capture_serverless` decorator on a lambda inside a Step Function that processes the output of a Map state.

Elastic APM Python package version: 6.14.0 (latest)

**To Reproduce**

1. Implement a Step Function with a Map state followed by a lambda function that takes the output of the map. As specified in AWS docs: `When complete, the output of the Map state is a JSON array, where each item is the output of an iteration.` - https://docs.aws.amazon.com/step-functions/latest/dg/concepts-asl-use-map-state-inline.html#:~:text=the%20ItemSelector%20field.-,When%20complete%2C%20the%20output%20of%20the%20Map%20state%20is%20a%20JSON%20array%2C%20where%20each%20item%20is%20the%20output%20of%20an%20iteration.,-For%20more%20information 
2. Add the `capture_serverless` setup to your lambda.

Expected: lambda to run and log to APM Server

Actual: Error from Step Function logs:
```
{
    "errorMessage": "'list' object has no attribute 'get'",
    "errorType": "AttributeError",
    "stackTrace": [
      "  File \"/var/task/elasticapm/contrib/serverless/aws.py\", line 116, in decorated\n    with _lambda_transaction(func, name, client, event, context) as sls:\n",
      "  File \"/var/task/elasticapm/contrib/serverless/aws.py\", line 147, in __enter__\n    trace_parent = TraceParent.from_headers(self.event.get(\"headers\", {}))\n"
    ]
  }
```

I think the fix is as simple as updating `elasticapm.xxx.aws`  `_lambda_transaction` to accept `event` being either a dictionary or a list, as this is an acceptable input to lambdas within Step Functions.

**Environment (please complete the following information)**
- OS: [e.g. Linux] - AWS Lambda function for Python
- Python version: 3.8
- APM Server version: 6.14.0

**Severity**
- Critical issue as this breaks all our lambdas processing events coming from Map states in Step Functions.

**Comment by basepi** [comment]
Thanks for the report! We'll get this fixed.

**Comment by basepi** [comment]
Fixed in #1775, thank you again for the report!
